### DISCUSSION TITLE
Open a sample project in the OpenROAD GUI

### DISCUSSION THREAD

**Discussion by vic4key** [The-OpenROAD-Project/OpenROAD#2965] [Q&A] [answered]
Hi everybody,

I already build OpenROAD GUI to run on Ubuntu Linux. Now I'm looking for a sample project that can open directly via main menu `File > Open DB` in OpenROAD GUI (like the below image), I have spent a lot of time to find the way (reading read-me, watching video, etc) to do it, but I did not complete.
I found a data test at https://github.com/TILOS-AI-Institute/MacroPlacement/tree/main/Flows/NanGate45/ariane133 but seems it cannot open directly by OpenROAD and maybe need to do several pre-processing steps before can open in OpenROAD GUI.
Could you please help me to know how to do it or give me detail instructions to open this test data in the OpenROAD GUI?
Thanks.

![sample](https://user-images.githubusercontent.com/5672864/222308822-2fc0204a-440c-4593-a04b-a4e82f591073.png)

**Comment by QuantamHD** [comment]
Hi @vic4key,

It's not clear what's failing, and what you would like to see? It would also be helpful if you uploaded the data you could upload your test data in a zip folder and post it here. So that we can debug th issue.

**Reply by vic4key** [reply]
Hi @QuantamHD, thanks for your reply.
The point here is I don't know how to open a test data in OpenROAD GUI via the main menu `File > Open DB`.
The expected that I would like to see is I can open a test data and display it in OpenROAD like the below image.
And the data you can get it here ["NanGate45/ariane133"](https://github.com/TILOS-AI-Institute/MacroPlacement/tree/main/Flows/NanGate45/ariane133) (as my understanding this is a test data for OpenROAD).

![sample](https://user-images.githubusercontent.com/5672864/222308822-2fc0204a-440c-4593-a04b-a4e82f591073.png)

**Reply by QuantamHD** [reply]
Have you run the flow following these instructions? https://github.com/TILOS-AI-Institute/MacroPlacement/tree/main/Flows#using-openroad-flow-scripts

The "data" in that repo includes the files you need to run the flow, but doesn't include the artifacts of the actual run. You need to run the flow first.

**Reply by vic4key** [reply]
Yes. Let me try following these steps to run the flow.

**Comment by vijayank88** [comment, marked as answer]
@vic4key 
Same design implemented in OpenROAD-flow-scripts. https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/tree/master/flow/designs/nangate45/ariane136
All you need to install OpenROAD-flow-scripts and run the design and view it in OpenROAD GUI.
Installation document available here: https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildLocally.html

**Reply by vic4key** [reply]
Thank you. I'm following the instruction that @QuantamHD https://github.com/The-OpenROAD-Project/OpenROAD/discussions/2965#discussioncomment-5174215 and you mentioned.
It's still running and seems takes so long time to finish (I built without enabled Nvidia CUDA, because I'm using VMware).
I don't know how long it takes. I hope it will end soon.

![20230302_190353](https://user-images.githubusercontent.com/5672864/222423805-01a8f413-a436-4717-8861-9432c33af83e.png)

**Comment by maliberty** [comment]
cuda will make almost no difference to runtime.  I would start by running gcd as it is the smallest test case.

**Reply by vic4key** [reply]
Hi @maliberty, @QuantamHD, and @QuantamHD I would like to say thank you so much for helping me.
I have succeeded to do this (although, it took a long time, so I let it run overnight 😁).
This is the result that I ran flow with the design ["NanGate45/ariane133"](https://github.com/TILOS-AI-Institute/MacroPlacement/tree/main/Flows/NanGate45/ariane133).

![20230303_101812](https://user-images.githubusercontent.com/5672864/222624089-5943f557-b825-48c8-8999-d6e7fb1ea4e8.png)

But I have some note, maybe you need, so I just write more. I'm using Ubuntu 22.04 and during working on it, I got 2 issues (not sure that it occurred in all machines or only mine), they are:
1. Missing a required package named `Klayout 0.27.1` (apt is missing this version, the default version in apt is 0.26.2). It can be found here https://www.klayout.de/build.html
2. At the finish step, this command is error, look at the below image: ![20230303_100628](https://user-images.githubusercontent.com/5672864/222625764-e3c28c2d-8949-47e3-81f1-ffdde297eece.png)

Finally, I have done now. Nice day bro(es).
